hydrolib/core/io/ini/models.py

```diff
@@ -153,74 +48,126 @@ class Comments(BaseModel, ABC):
 
         class Config:
             extra = Extra.allow
-            allow_population_by_field_name = True
             arbitrary_types_allowed = False
 
     comments: Optional[Comments] = None
 
-    @validator("comments", always=True)
+    @validator("comments", always=True, allow_reuse=True)
     def comments_matches_has_comments(cls, v):
-        if cls._supports_comments() and v is None:
-            raise ValueError(f"{cls} should have comments.")
-        elif not cls._supports_comments() and v is not None:
-            raise ValueError(f"{cls} should not have comments.")
-
+        if not cls._supports_comments() and v is not None:
+            logging.warning(f"Dropped unsupported comments from {cls.__name__} init.")
+            v = None
         return v
 
     @classmethod
     def validate(cls: Type["IniBasedModel"], value: Any) -> "IniBasedModel":
         if isinstance(value, Section):
-            # value = value.flatten()  # TODO
-            value = cls._convert_section(value)
+            value = value.flatten(
+                cls._duplicate_keys_as_list(), cls._supports_comments()
+            )
 
         return super().validate(value)
 
-    @classmethod
-    def _convert_section(cls, section: Section) -> Dict:
-        converted_content = cls._convert_section_content(section.content)
-        underlying_dict = cls._convert_section_to_dict(section)
-        return {**underlying_dict, **converted_content}
+
+class DataBlockIniBasedModel(IniBasedModel):
+    """DataBlockIniBasedModel defines the base model for ini models with datablocks."""
 
     @classmethod
     def _convert_section_to_dict(cls, value: Section) -> Dict:
         return value.dict(
             exclude={
                 "start_line",
                 "end_line",
-                "datablock",
                 "content",
             }
         )
 
+
+class INIGeneral(IniBasedModel):
```

Comment:
Shouldn't these and the subsequent classes be in a different file? I feel like they refer to some more specific objects, though I'm not sure which ones.

Reply:
This is the header part of the `.ini` files, in which eventual version checks and warnings can be stored. Should be in `models`, like the rest of the representation of these files?